### PR TITLE
fix(rust): add error handling logic for std duration; add support for chrono duration

### PR DESCRIPTION
**Pull request by BaldDemian**
## Why?

The default duration type in Rust is `std::time::Duration`, which is unsigned, causing potential problems mentioned in https://github.com/apache/fory/issues/3484.

## What does this PR do?

For compatibility purposes, code related to `std::time::Duration` is not removed, instead I add some error handling logic around it.

I add write/read logic for `chrono::Duration` similar to `std::time::Duration`. It is worth noting that the maximum value of `seconds` in `chrono::Duration` is actually `i64::MAX/100`, smaller than what the spec allows, but I think this is fine due to the following reasons:
1. `i64::MAX/100` is already large enough for real use cases
2. duration types in C++, C# module in Fory also have maximum limits less than `i64::MAX`

## Related issues
Close https://github.com/apache/fory/issues/3484

## AI Contribution Checklist

No

## Does this PR introduce any user-facing change?

No. Code related to `std::time::Duration` is not removed

## Benchmark

Existing benchmark code doesn't cover `std::time::Duration`.

### CODE REVIEW

File: rust/fory-core/src/serializer/datetime.rs [hunk 1]
```diff
@@ -243,4 +340,52 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn test_chrono_duration_serialization() {
+        let fory = Fory::default();
+
+        // Test various durations
+        let test_cases = vec![
```
Comment (chaokunyang) [inline rust/fory-core/src/serializer/datetime.rs:349]: could you add some negative duration here? and also add some edge cases such as max/min allowed values

Comment (BaldDemian) [reply rust/fory-core/src/serializer/datetime.rs:349]: done

**Comment by BaldDemian** [comment]
BTW, at first I didn't notice that there is no benchmark case covering Duration so I ran the whole `cargo bench`. When hitting the `8_threads_shared_fory` case, error occurred. The error can be stably reproduced in `benchmarks/rust` and run `cargo -q bench --bench threaded_bench -- "ThreadedSerialization/8_threads_shared_fory" --exact --noplot`

Screenshot 2026-03-18 at 12 11 19 AM

I may spend some time finding out why.

**Review by chaokunyang — APPROVED** [review]
LGTM
